utils.js

```diff
@@ -4759,11 +4759,89 @@ export async function parseCargoTomlData(cargoTomlFile, simple = false) {
   }
 }
 
-export async function parseCargoData(cargoData) {
+/**
+ * Parse a Cargo.lock file to find components within the Rust project.
+ *
+ * @param {string} cargoLockFile A path to a Cargo.lock file. The Cargo.lock-file path may be used as information for extended attributes, such as manifest based evidence.
+ * @param {boolean} simple Return a simpler representation of the component by skipping extended attributes and license fetch.
+ *
+ * @returns {array} A list of the project's components as described by the Cargo.lock-file.
+ */
+export async function parseCargoData(cargoLockFile, simple = false) {
+  const addPackageToList = (packageList, newPackage, { simple }) => {
+    if (!newPackage) {
+      return;
+    }
+
+    const purl = new PackageURL(
+      "cargo",
+      "",
+      newPackage.name,
+      newPackage.version,
+      null,
+      null
+    ).toString();
+    const component = {
+      type: "library",
+      group: newPackage.group,
+      "bom-ref": purl,
```

Comment:
bom-ref is usually `decodeURIComponent(purl)`. But for cargo this is fine since the group is empty and name doesn't have slash characters.